Sources/NIO/Bootstrap.swift

```diff
@@ -364,8 +385,121 @@ public final class ServerBootstrap {
+                case .autoRead(let value):
+                    _ = serverBootstrap.serverChannelOption(ChannelOptions.autoRead, value: value)
+                case .reuseAddr(let value):
+                    _ = serverBootstrap.serverChannelOption(ChannelOptions.socketOption(.reuseaddr),
+                                                            value: value ? 1 : 0)
+                case .backlog(let value):
+                    _ = serverBootstrap.serverChannelOption(ChannelOptions.backlog, value: value)
+                }
+            }
+        }
+    }
+    
+    /// A channel option which can be applied to bootstrap using shorthand notation.
+    /// - See: ServerBootstrap.serverOptions(_ options: [ShorthandBootstrapOption])
+    public struct ShorthandChildBootstrapOption {
+        private let data: ShorthandChildOption
+        
+        private init(_ data: ShorthandChildOption) {
+            self.data = data
+        }
+        
+        /// Apply the contained option to the supplied ServerBootstrap
+        /// - Parameter serverBootstrap: bootstrap to apply this option to.
+        /// - Returns: the modified bootstrap (currently the same one mutated)
+        func applyOption(to serverBootstrap: ServerBootstrap) {
+            data.applyOption(to: serverBootstrap)
+        }
+        
+        fileprivate enum ShorthandChildOption {
+            case reuseAddr(Bool)
+            case allowRemoteHalfClosure(Bool)
+            
+            func applyOption(to serverBootstrap: ServerBootstrap) {
+                switch self {
+                case .reuseAddr(let value):
+                    _ = serverBootstrap.childChannelOption(ChannelOptions.socketOption(.reuseaddr),
+                                                           value: value ? 1 : 0)
+                case .allowRemoteHalfClosure(let value):
+                    _ = serverBootstrap.childChannelOption(ChannelOptions.allowRemoteHalfClosure, value: value)
+                }
+            }
+        }
+    }
+}
+
+// Hashable for the convenience of users.
+extension ServerBootstrap.ShorthandServerBootstrapOption : Hashable {}
+extension ServerBootstrap.ShorthandServerBootstrapOption.ShorthandServerOption : Hashable {}
+
+/// Approved shorthand server options.
+extension ServerBootstrap.ShorthandServerBootstrapOption {
+    /// Option to reuse address.
+    /// - See:  NIOBSDSocket.Option.reuseaddr
+    public static let reuseAddr = ServerBootstrap.ShorthandServerBootstrapOption(.reuseAddr(true))
+    
+    /// Option to disable autoRead
+    /// - See: ChannelOptions.autoRead
+    public static let disableAutoRead = ServerBootstrap.ShorthandServerBootstrapOption(.autoRead(false))
+    
+    /// `BacklogOption` allows users to configure the `backlog` value as specified in `man 2 listen`.
+    /// This is only useful for `ServerSocketChannel`s.
+    /// - See: ChannelOptions.backlog
+    public static func backlog(_ value: ChannelOptions.Types.BacklogOption.Value) ->
```

Comment:
I think we should add more words here too. Maybe `.maximumUnacceptedConnectionBacklog` ? It's a terrible name but we should clarify what is it.

@Lukasa what does Network.framework support here?

Reply:
Network.framework doesn't have an exact analog of this field. The closest it has is [`NWListener.newConnectionLimit`](https://developer.apple.com/documentation/network/nwlistener/3200460-newconnectionlimit) which is not really the same, it's more like Combine's demand.

Reply:
@Lukasa thanks, that's interesting. This sounds like backlog except that it doesn't seem like a "hint" to the OS but rather a setting that Network.framework will actually respect in all cases.
The other difference is the default: Normal Unixes default to "whatever, probably a small power of two, maybe 16?" whereas Network.framework seems to default to infinitely many outstanding...

Reply:
Correct, this was put in place in part to facilitate NIO's desire to have `read` be semantically meaningful on server channels, so it needs to be actually obeyed. It's much more like "controlling how much you call `accept`" than it is like `backlog`.